app/services/wrappers/ltsa_parcel_map_bc.rb

```diff
@@ -0,0 +1,158 @@
+    }
+
+    get("#{PARCEL_SERVICE}/query", query_params, true)
+  end
+
+  def historic_site_by_pid(pid:)
+    query_params = {
+      f: "json",
+      where: "HISTORIC_SITE_IND='Y' AND PARCEL_DESCRIPTION='#{pid}'",
+      returnGeometry: true,
+      outFields: "*",
+    }
+    response = get("#{HISTORIC_SERVICE}/query", query_params, true)
+
+    # Assume if there is a parcel description match not to use LTSA geometry matching
+    if response.success? && JSON.parse(response.body).dig("features").length > 0
+      return parse_attributes_from_response(response)
+    end
+
+    # Get geometry from PID
+    response_for_geometry = get_details_by_pid(pid: pid, fields: "PID")
+    if response_for_geometry.success? && JSON.parse(response_for_geometry.body).dig("features", 0, "geometry")
+      # Intersect the geometry from the city over
+      pid_geo = JSON.parse(response_for_geometry.body).dig("features", 0, "geometry")
+      pid_geo_query_params = {
+        f: "json",
+        where: "HISTORIC_SITE_IND='Y'",
+        returnGeometry: true,
+        spatialRel: "esriSpatialRelIntersects",
+        geometry: pid_geo.to_json,
+        geometryType: "esriGeometryPolygon",
+        outFields: "*",
+      }
+
+      pid_geo_response = get("#{HISTORIC_SERVICE}/query", pid_geo_query_params, true)
+      return parse_attributes_from_response(pid_geo_response)
+    else
+      raise Errors::GeometryError
+    end
+  end
+
+  def get_dynamic_features_by_pid(pid:)
+    #call the get_details_by_pid to get the active polygon for the parcel
+    query_params = { f: "json", returnIdsOnly: false, returnCountOnly: false }
+
+    get("#{PARCEL_SERVICE}/query", query_params, true)
+
+    # https://maps.gov.bc.ca/arcserver/rest/services/mpcm/bcgw/MapServer/dynamicLayer/query?layer=Your_Layer_Definition&f=json&returnGeometry=true&spatialRel=esriSpatialRelIntersects&geometry={"rings":[[[x1, y1], [x2, y2], [x3, y3], ..., [x1, y1]]],"spatialReference":{"wkid":Spatial_Reference_ID}}&geometryType=esriGeometryPolygon&inSR=Spatial_Reference_ID&outSR=Spatial_Reference_ID
+  end
+
+  def get_feature_attributes_by_pid(pid:, fields: "*")
+    response = get_details_by_pid(pid: pid, fields: fields)
+    if response.success?
+      return parse_attributes_from_response(response)
+    else
+      raise Errors::FeatureAttributesRetrievalError
+    end
+  end
+
+  def get_feature_attributes_by_pid_or_pin(pid:, pin:, fields: "*")
+    raise Errors::FeatureAttributesRetrievalError if pin.blank? && pid.blank?
+    #always prioritize pid over pin
+    response =
+      if pid
+        get_details_by_pid(pid: pid, fields: fields)
+      elsif pin
+        get_details_by_pin(pin: pin, fields: fields)
+      end
+    if response.success?
+      return parse_attributes_from_response(response)
+    else
+      raise Errors::FeatureAttributesRetrievalError
+    end
+  end
+
+  def get_coordinates_by_pid(pid)
+    response = get_details_by_pid(pid: pid)
+    if response.success?
```

Comment:
if the base `get` method already raises an error's for non 200 responses wouldn't the else condition never execute?

Reply:
the base offers the ability to skip handling the response, as to accommodate the way Sunny set up his integrations class. skip handle response is passed as true when more detailed control over the response is needed.